ddtrace/profiling/collector/_memalloc_heap.c

```diff
@@ -21,6 +27,17 @@ typedef struct
 
 static heap_tracker_t global_heap_tracker;
 
+
+static uint32_t
+heap_tracker_next_sample_size(uint32_t sample_size)
+{
+    // 0x4000000 == pow(2, 26)
+    uint32_t q = random_range(0x4000000);
+    double log_val = log2(q) - 26;
```

Comment:
Isn't this the same as
~~~ c
double log_val = log2(random())
~~~
where `random` is
~~~ c
static double random(void) {
  return (double)rand() / ((double)RAND_MAX + 1)
}
~~~
This would save a multiplication by `max`

Reply:
Yeah I think that should work.